prefect2_api/block_document.go

```diff
@@ -0,0 +1,121 @@
+package prefect2_api
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func (c *Client) GetAllBlockDocuments(workspaceID string) ([]BlockDocument, error) {
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/accounts/%s/workspaces/%s/block_documents/filter", c.PrefectApiUrl, c.PrefectAccountId, workspaceID), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req, c.PrefectApiKey)
+	if err != nil {
+		return nil, err
+	}
+
+	blockDocuments := []BlockDocument{}
+
+	err = json.Unmarshal(body, &blockDocuments)
+	if err != nil {
+		return nil, err
+	}
+
+	return blockDocuments, nil
+}
+
+func (c *Client) GetBlockDocument(blockDocumentId string, workspaceID string) (*BlockDocument, error) {
```

Comment:
Recent versions of Go now pass `ctx.Context` parameter as the first parameter by convention, because they are required to support timeouts and request cancellation. We can take care of it after merging, so it's not a big deal, but I think this function would be more idiomatic if they looked like this:

```suggestion
func (c *Client) GetBlockDocument(ctx context.Context, blockDocumentId string, workspaceID string) (*BlockDocument, error) {
	req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("%s/accounts/%s/workspaces/%s/block_documents/%s", c.PrefectApiUrl, c.PrefectAccountId, workspaceID, blockDocumentId), nil)
```

See: https://pkg.go.dev/net/http#NewRequestWithContext and https://go.dev/blog/context

Reply:
Awesome, I've updated all functions inside the `_api` folder